src/main/java/com/marklogic/kafka/connect/source/DslQueryHandler.java

```diff
@@ -29,27 +30,29 @@ public DslQueryHandler(DatabaseClient databaseClient, Map<String, Object> parsed
         } else {
             constraintColumnName = QueryHandlerUtil.sanitize(rawConstraintColumnName);
         }
+        rowLimit = (Integer) parsedConfig.get(MarkLogicSourceConfig.ROW_LIMIT);
     }
 
     @Override
     public PlanBuilder.Plan newPlan(String previousMaxConstraintColumnValue) {
-        currentDslQuery = injectConstraintIntoQuery(previousMaxConstraintColumnValue);
+        currentDslQuery = appendConstraintAndOrderByToQuery(previousMaxConstraintColumnValue);
+        if (rowLimit > 0) {
+            currentDslQuery = appendLimitToQuery(currentDslQuery);
+        }
         logger.info("DSL query: " + currentDslQuery);
         return databaseClient.newRowManager().newRawQueryDSLPlan(new StringHandle(currentDslQuery));
     }
 
-    protected String injectConstraintIntoQuery(String previousMaxConstraintColumnValue) {
+    protected String appendLimitToQuery(String currentDslQuery) {
+        return currentDslQuery + ".limit(" + rowLimit + ")";
+    }
+
+    protected String appendConstraintAndOrderByToQuery(String previousMaxConstraintColumnValue) {
         String constrainedDsl = userDslQuery;
         if (previousMaxConstraintColumnValue != null) {
-            String constraintPhrase = ".where(op.gt(op.col(\"" + constraintColumnName + "\"), \"" + previousMaxConstraintColumnValue + "\"))";
-            String originalDslNoWhitespace = QueryHandlerUtil.removeWhitespace(userDslQuery);
-            if (originalDslNoWhitespace.contains(").")) {
-                int firstClosingParen = originalDslNoWhitespace.indexOf(").");
-                constrainedDsl = originalDslNoWhitespace.substring(0, firstClosingParen + 1)
-                    + constraintPhrase + originalDslNoWhitespace.substring(firstClosingParen + 1);
-            } else {
-                constrainedDsl = userDslQuery + constraintPhrase;
-            }
+            String constraintPhrase = ".where(op.gt(op.col('" + constraintColumnName + "'), '" + previousMaxConstraintColumnValue + "'))"
```

Comment:
Just occurred to me that we can use bindParam, as that's exposed on the RawPlan interface. That allows us to avoid string concatenation of values (but as mentioned in chat, doesn't seem possible yet to avoid string concatenation entirely). I think this should do the trick:

```
PlanBuilder.Plan plan = databaseClient.newRowManager().newRawQueryDSLPlan(new StringHandle(currentDslQuery));
        if (rowLimit > 0) {
            plan.bindParam("rowLimit", rowLimit);
        }
        if (previousMaxConstraintColumnValue != null) {
            plan.bindParam("previousMaxConstraintColumnValue", previousMaxConstraintColumnValue);
        }
        return plan;
```

That should work after modifying the query to do something like this:

    return currentDslQuery + ".limit(op.param('rowLimit'))";